packages/nix/mkDevShells/pf-env-bootstrap-restore.sh

```diff
@@ -0,0 +1,255 @@
+    kubectl "$CONTEXT_ARGS" patch nodepool.karpenter.sh "$nodepool" -n "karpenter" --type='json' -p='[{"op": "remove", "path": "/spec/limits"}]'
+  done
+}
+
+clear_pending_pods() {
+  echo "Listing all Pending pods..."
+
+  # Get all pods in 'Pending' state across all namespaces
+  pending_pods=$(kubectl "$CONTEXT_ARGS" get pods --all-namespaces --field-selector=status.phase=Pending -o json | jq -r '.items[] | "\(.metadata.namespace) \(.metadata.name)"')
+
+  if [ -z "$pending_pods" ]; then
+    echo "No Pending pods found."
+  else
+    echo "Found Pending pods, starting deletion process..."
+
+    # Loop through each pending pod and delete it
+    while read -r namespace pod_name; do
+      echo "Deleting pod: $pod_name in namespace: $namespace"
+      kubectl "$CONTEXT_ARGS" delete pod "$pod_name" -n "$namespace"
+    done <<<"$pending_pods"
+
+    echo "All Pending pods have been deleted."
+  fi
+}
+
+wait_for_cilium_agent() {
+  local namespace=$1
+  local deployment_name="cilium-agent"
+
+  echo "Waiting for the scheduler pod to be scheduled and running in namespace $namespace..."
+
+  # Timeout after 5 minutes (300 seconds)
+  timeout=300
+  interval=10
+  elapsed=0
+
+  while [[ $elapsed -lt $timeout ]]; do
+    pod_status=$(kubectl "$CONTEXT_ARGS" get pod -n "$namespace" -l panfactum.com/workload=scheduler -o jsonpath='{.items[0].status.phase}' 2>/dev/null)
+
+    if [[ $pod_status == "Running" ]]; then
+      echo "$deployment_name pod is running."
+      return 0
+    else
+      echo "Pod status: $pod_status. Waiting..."
+    fi
+
+    sleep $interval
+    elapsed=$((elapsed + interval))
+  done
+
+  echo "Timed out waiting for $deployment_name pod to be scheduled."
+  return 1
+}
+
+restore_kube_config() {
+  restore_kube_config_path=$(get_cluster_state "restore_kubeconfig")
+
+  if [[ -f $restore_kube_config_path ]]; then
+    echo "Restoring kubeconfig from $restore_kube_config_path"
+    cp -f "$restore_kube_config_path" "$KUBE_DIR/config"
+    mv "$restore_kube_config_path" "$restore_kube_config_path-restored"
+  fi
+}
+
+clear_cluster_state() {
+  echo "Clearing cluster state for $EKS_CLUSTER_NAME file... $STATE_FILE"
+  jq --arg cluster "$EKS_CLUSTER_NAME" 'del(.[$cluster])' "$STATE_FILE" >"$STATE_FILE.tmp" && mv "$STATE_FILE.tmp" "$STATE_FILE"
+  echo "Cleared $EKS_CLUSTER_NAME state file."
+}
+
+restore() {
+  # Scale up 'nat-' ASGs to 1
+  scale_asgs_nat_nodes 1 1
+
+  # update the cilium operator deployment to use the `default-scheduler`
+  update_scheduler "cilium" "cilium-operator" "default-scheduler"
+
+  # clear pending pods
+  clear_pending_pods
+
+  # Scale up EKS controller ASGs based on tags
+  controller_node_count=$(get_cluster_state "restore_controller_node_count")
+  scale_asg_controller_nodes "$controller_node_count" "$controller_node_count"
+
+  # Scale up Karpenter node pools
+  scale_up_node_pools
+
+  # Wait for Controller pods to scale
+  wait_for_cilium_agent "scheduler"
```

Comment:
This doesn't make sense.

Reply:
only perform if panfactum scheduler deployment exists